### PR TITLE
Adding app resource

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/6654

Copying `app` resource from giantswarm/release-operator. 

Origin: https://github.com/giantswarm/release-operator/tree/master/pkg/controller/resource/app

We need `app` resource to migrate `chartoperator`  responsibility to cluster-operator.

### CODE REVIEW

File: pkg/resource/app/create.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
```
Comment (tomahawk28) [inline pkg/resource/app/create.go]: `apiextensions` is a new home for this resource, moving imports as appropriate way.

File: Gopkg.toml [hunk 2]
```diff
@@ -65,3 +65,7 @@ required = [
 [[prune.project]]
     name = "k8s.io/gengo"
     unused-packages = false
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/giantswarm/microerror"
```
Comment (rossf7) [inline Gopkg.toml]: This is in the wrong place and operatorkit needs to be added.

File: pkg/resource/app/create.go (hunk 1 above)
Comment (kopiczko) [reply pkg/resource/app/create.go]: New home :D

File: Gopkg.toml [hunk 3]
```diff
@@ -65,3 +73,4 @@ required = [
 [[prune.project]]
     name = "k8s.io/gengo"
     unused-packages = false
+
```
Comment (kopiczko) [inline Gopkg.toml]:
```suggestion
```

**Review by kopiczko — APPROVED** [review]
LGTM - I didn't check the code but I understand it remained unchanged

File: Gopkg.toml (hunk 2 above)
Comment (tomahawk28) [reply Gopkg.toml]: @rossf7 Thanks, I fixed it.

File: pkg/resource/app/create.go (hunk 1 above)
Comment (rossf7) [reply pkg/resource/app/create.go]: New home is nice! ❤️

**Pull request merged**